src/languages/snowflake/snowflake.functions.ts

```diff
@@ -0,0 +1,614 @@
+    'ST_GEOGRAPHYFROMWKT',
+    'ST_GEOHASH',
+    'ST_GEOMETRYFROMWKB',
+    'ST_GEOMETRYFROMWKT',
+    'ST_HAUSDORFFDISTANCE',
+    'ST_INTERSECTION',
+    'ST_INTERSECTS',
+    'ST_LENGTH',
+    'ST_MAKEGEOMPOINT',
+    'ST_GEOM_POINT',
+    'ST_MAKELINE',
+    'ST_MAKEPOINT',
+    'ST_POINT',
+    'ST_MAKEPOLYGON',
+    'ST_POLYGON',
+    'ST_NPOINTS',
+    'ST_NUMPOINTS',
+    'ST_PERIMETER',
+    'ST_POINTN',
+    'ST_SETSRID',
+    'ST_SIMPLIFY',
+    'ST_SRID',
+    'ST_STARTPOINT',
+    'ST_SYMDIFFERENCE',
+    'ST_UNION',
+    'ST_WITHIN',
+    'ST_X',
+    'ST_XMAX',
+    'ST_XMIN',
+    'ST_Y',
+    'ST_YMAX',
+    'ST_YMIN',
+    'STAGE_DIRECTORY_FILE_REGISTRATION_HISTORY',
+    'STAGE_STORAGE_USAGE_HISTORY',
+    'STARTSWITH',
+    'STDDEV',
+    'STDDEV_POP',
+    'STDDEV_SAMP',
+    'STRIP_NULL_VALUE',
+    'STRTOK',
+    'STRTOK_SPLIT_TO_TABLE',
+    'STRTOK_TO_ARRAY',
+    'SUBSTR',
+    'SUBSTRING',
+    'SUM',
+    'SYSDATE',
+    'SYSTEM$ABORT_SESSION',
+    'SYSTEM$ABORT_TRANSACTION',
+    'SYSTEM$AUTHORIZE_PRIVATELINK',
+    'SYSTEM$AUTHORIZE_STAGE_PRIVATELINK_ACCESS',
+    'SYSTEM$BEHAVIOR_CHANGE_BUNDLE_STATUS',
+    'SYSTEM$CANCEL_ALL_QUERIES',
+    'SYSTEM$CANCEL_QUERY',
+    'SYSTEM$CLUSTERING_DEPTH',
+    'SYSTEM$CLUSTERING_INFORMATION',
+    'SYSTEM$CLUSTERING_RATIO ',
+    'SYSTEM$CURRENT_USER_TASK_NAME',
+    'SYSTEM$DATABASE_REFRESH_HISTORY ',
+    'SYSTEM$DATABASE_REFRESH_PROGRESS',
+    'SYSTEM$DATABASE_REFRESH_PROGRESS_BY_JOB ',
+    'SYSTEM$DISABLE_BEHAVIOR_CHANGE_BUNDLE',
+    'SYSTEM$DISABLE_DATABASE_REPLICATION',
+    'SYSTEM$ENABLE_BEHAVIOR_CHANGE_BUNDLE',
+    'SYSTEM$ESTIMATE_QUERY_ACCELERATION',
+    'SYSTEM$ESTIMATE_SEARCH_OPTIMIZATION_COSTS',
+    'SYSTEM$EXPLAIN_JSON_TO_TEXT',
+    'SYSTEM$EXPLAIN_PLAN_JSON',
+    'SYSTEM$EXTERNAL_TABLE_PIPE_STATUS',
+    'SYSTEM$GENERATE_SAML_CSR',
+    'SYSTEM$GENERATE_SCIM_ACCESS_TOKEN',
+    'SYSTEM$GET_AWS_SNS_IAM_POLICY',
+    'SYSTEM$GET_PREDECESSOR_RETURN_VALUE',
+    'SYSTEM$GET_PRIVATELINK',
+    'SYSTEM$GET_PRIVATELINK_AUTHORIZED_ENDPOINTS',
+    'SYSTEM$GET_PRIVATELINK_CONFIG',
+    'SYSTEM$GET_SNOWFLAKE_PLATFORM_INFO',
+    'SYSTEM$GET_TAG',
+    'SYSTEM$GET_TAG_ALLOWED_VALUES',
+    'SYSTEM$GET_TAG_ON_CURRENT_COLUMN',
+    'SYSTEM$GET_TAG_ON_CURRENT_TABLE',
+    'SYSTEM$GLOBAL_ACCOUNT_SET_PARAMETER',
+    'SYSTEM$LAST_CHANGE_COMMIT_TIME',
+    'SYSTEM$LINK_ACCOUNT_OBJECTS_BY_NAME',
+    'SYSTEM$MIGRATE_SAML_IDP_REGISTRATION',
+    'SYSTEM$PIPE_FORCE_RESUME',
+    'SYSTEM$PIPE_STATUS',
+    'SYSTEM$REVOKE_PRIVATELINK',
+    'SYSTEM$REVOKE_STAGE_PRIVATELINK_ACCESS',
+    'SYSTEM$SET_RETURN_VALUE',
+    'SYSTEM$SHOW_OAUTH_CLIENT_SECRETS',
+    'SYSTEM$STREAM_GET_TABLE_TIMESTAMP',
+    'SYSTEM$STREAM_HAS_DATA',
+    'SYSTEM$TASK_DEPENDENTS_ENABLE',
+    'SYSTEM$TYPEOF',
+    'SYSTEM$USER_TASK_CANCEL_ONGOING_EXECUTIONS',
+    'SYSTEM$VERIFY_EXTERNAL_OAUTH_TOKEN',
+    'SYSTEM$WAIT',
+    'SYSTEM$WHITELIST',
+    'SYSTEM$WHITELIST_PRIVATELINK',
```

Comment:
Currently these won't work because the functions list gets concatenated to a regex and `$` will attempt to match end-of-line.

I should fix this by adding escaping of function names before constructing a regex.